bboxes labelling/extract_frames_to_label_w_sleap.py

```diff
@@ -192,27 +325,136 @@ def extract_frames_to_label(args):
                     logging.info(f"frame {frame_idx} saved at {file_path}")
                 else:
                     logging.info(
-                        f"ERROR saving {Path(vid_str).stem},"
-                        f" frame {frame_idx}...skipping",
+                        f"ERROR saving {Path(vid_str).stem}, frame {frame_idx}"
+                        "...skipping"
                     )
                     continue
 
-        # close capture
+        # close video capture
         cap.release()
 
 
-if __name__ == "__main__":
-    # ------------------------------------
-    # parse command line arguments
-    # ------------------------------
+def compute_and_extract_frames_to_label(args):
+    """Compute suggested frames to label and
+    extract them as png files.
+
+    We use SLEAP's image feature method to select
+    the frames for labelling and export them as png
+    files in the desired directory.
+    We also output to the same location the list of
+    frame indices selected per video as a json file.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        data structure holding parsed input arguments.
+        It includes the following attributes:
+            'list_video_locations',
+            'output_path',
+            'video_extensions',
+            'initial_samples',
+            'sample_method',
+            'scale',
+            'feature_type',
+            'n_components', '
+            n_clusters',
+            'per_cluster',
+            'compute_features_per_video'
+    """
+
+    # Compute list of suggested frames using SLEAP
+    map_videos_to_extracted_frames = compute_suggested_sleap_frames(
```

Comment:
thanks for raising this point, I realised I often have this doubt....

One the one hand I find that explicitly showing the parameters in the function signature makes it more readable (you see at a glance what matters to the method), but you are write that it can become long and harder to maintain.

I read a bit more about this to see what are the trade-offs. In case you are interested, I found this [SO post](https://stackoverflow.com/a/36905418) and [this refactoring blog post](https://refactoring.guru/introduce-parameter-object) quite nice.

From what I understand it seems that using a parameter object (instead of passing the naked parameters) is mostly beneficial when (1) the parameters get passed around many times, and (2) when it makes sense to include methods in the class that holds the parameters, beyond setter/getter methods. Making a class just holding parameters is what in the refactoring blog they call a ["data class code smell"](https://refactoring.guru/smells/data-class).

Since in this case we are only passing these parameters once, and we don't really need as of now further methods beyond setter/getter ones, I would say we leave it with the 'naked parameters' option. But I am happy to keep an eye on it.... I don't use OOP approaches much, so I am a bit biased towards 'pure function' approaches

Reply:
Yes there is always pro and cons on any method. Yes we can keep it as it is for now as the code is not huge and we only pass them once. Thank you for sharing the links!